File: 力扣-练习/数组/26. 删除有序数组中的重复项.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int size = nums.size();
        if(size < 2) return size;
        int left=0, right=1;
        while(right < size)
        {
            if(nums[left] == nums[right])
            {
                right++;
            }
            else
            {
                // left ++;
                nums[++left] = nums[right++];
                // right ++;
            }
        }
        return (left+1);
    }
};

File: 力扣-练习/数组/26. 删除有序数组中的重复项.py
"""
# !/usr/bin/env python
# -*- coding: utf-8 -*-
@Time        : 2022/5/12 10:34
@File        : 26. 删除有序数组中的重复项.py
"""


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = len(nums)

        if l < 2:
            return l

        j = 0
        for i in range(1, l):
            if nums[i] != nums[j]:
                j += 1
                nums[j] = nums[i]
        return j + 1
